### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by dgeare**
Adds tests for text, position, box shadow, and flex tools.

### CODE REVIEW

File: app/features/position.test.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  await page.click(test_selector)
+  await page.keyboard.down("Shift")
+  await page.keyboard.press('ArrowUp')
+  const changedPageTop = await page.$eval(test_selector, el => el.getBoundingClientRect().top)
+  t.true(originalPageTop - 10 == changedPageTop)
+  t.pass()
+})
+
+
+test('Test Drag Works', async t => {
```
Comment (argyleink) [inline app/features/position.test.js]: aw snap!

File: app/features/boxshadow.test.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+const tool            = 'boxshadow'
+const test_selector   = '[intro] b'
+
+const getShadowValues = async (page, testEl = test_selector) => {
+  const shadowStr = await page.$eval(testEl, el => el.style.boxShadow)
+  return parseShadowValues(shadowStr)
+}
+
+const parseShadowValues = (str) => {
+    const [,color,x,y,blur,spread,inset] = /([^\)]+\)) ([^\s]+) ([^\s]+) ([^\s]+) ([^\s]+)( inset)?/.exec(str)
```
Comment (argyleink) [inline app/features/boxshadow.test.js]:
```suggestion
  const [,color,x,y,blur,spread,inset] = /([^\)]+\)) ([^\s]+) ([^\s]+) ([^\s]+) ([^\s]+)( inset)?/.exec(str)
```

File: app/features/boxshadow.test.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    return { color, x, y, blur, spread, inset : inset !== undefined }
```
Comment (argyleink) [inline app/features/boxshadow.test.js]:
```suggestion
  return { color, x, y, blur, spread, inset : inset !== undefined }
```

File: app/features/boxshadow.test.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  t.pass()
+})
+
+test('Can adjust X position', async t => {
+  const { page } = t.context
+
+  await page.click(test_selector)
+  await page.keyboard.press('ArrowRight')
+  let shadow = await getShadowValues(page)
+  t.true(shadow.x == "1px")
```
Comment (argyleink) [inline app/features/boxshadow.test.js]:
```suggestion
  t.true(shadow.x === "1px")
```

File: app/features/boxshadow.test.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  //test shift case
+  await page.keyboard.down('Shift')
+  await page.keyboard.press('ArrowRight')
+  shadow = await getShadowValues(page)
+  t.true(shadow.x == "11px")
```
Comment (argyleink) [inline app/features/boxshadow.test.js]:
```suggestion
  t.true(shadow.x === "11px")
```

File: app/features/boxshadow.test.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+})
+
+
+test('Can adjust Y position', async t => {
+  const { page } = t.context
+
+  await page.click(test_selector)
+  await page.keyboard.press('ArrowDown')
+  let shadow = await getShadowValues(page)
+  t.true(shadow.y == "1px")
```
Comment (argyleink) [inline app/features/boxshadow.test.js]:
```suggestion
  t.true(shadow.y === "1px")
```

File: app/features/boxshadow.test.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+  //test shift case
+  await page.keyboard.down('Shift')
+  await page.keyboard.press('ArrowDown')
+  shadow = await getShadowValues(page)
+  t.true(shadow.y == "11px")
```
Comment (argyleink) [inline app/features/boxshadow.test.js]:
```suggestion
  t.true(shadow.y === "11px")
```

File: app/features/boxshadow.test.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+test('Shadow Blur Works', async t => {
+  const { page } = t.context
+
+  await page.click(test_selector)
+  await page.keyboard.press('ArrowDown')
+  await page.keyboard.down('Alt')
+  await page.keyboard.press('ArrowUp')
+  let shadow = await getShadowValues(page)
+  t.true(shadow.blur == "1px")
```
Comment (argyleink) [inline app/features/boxshadow.test.js]:
```suggestion
  t.true(shadow.blur === "1px")
```

File: app/features/boxshadow.test.js [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+  //test shift case
+  await page.keyboard.down('Shift')
+  await page.keyboard.press('ArrowUp')
+  shadow = await getShadowValues(page)
+  t.true(shadow.blur == "11px")
```
Comment (argyleink) [inline app/features/boxshadow.test.js]:
```suggestion
  t.true(shadow.blur === "11px")
```

File: app/features/boxshadow.test.js [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+
+test('Shadow Spread Works', async t => {
+  const { page } = t.context
+
+  await page.click(test_selector)
+  await page.keyboard.press('ArrowDown')
+  await page.keyboard.down('Alt')
+  await page.keyboard.press('ArrowRight')
+  let shadow = await getShadowValues(page)
+  t.true(shadow.spread == "1px")
```
Comment (argyleink) [inline app/features/boxshadow.test.js]:
```suggestion
  t.true(shadow.spread === "1px")
```

File: app/features/boxshadow.test.js [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+  //test shift case
+  await page.keyboard.down('Shift')
+  await page.keyboard.press('ArrowRight')
+  shadow = await getShadowValues(page)
+  t.true(shadow.spread == "11px")
```
Comment (argyleink) [inline app/features/boxshadow.test.js]:
```suggestion
  t.true(shadow.spread === "11px")
```

File: app/features/boxshadow.test.js [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+  t.pass()
+})
+
+// test('Shadow can be set to inset', async t => {
```
Comment (argyleink) [inline app/features/boxshadow.test.js]: test works, comment back in

File: app/features/boxshadow.test.js [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+//   const { page } = t.context
+
+//   await page.click(test_selector)
+//   await page.keyboard.press('ArrowDown')
+//   await page.keyboard.down('Control')
```
Comment (argyleink) [inline app/features/boxshadow.test.js]: key "Meta" works on macOS. create a function that uses pptr to eval current OS and return the appropriate key

File: app/features/text.test.js [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+})
+
+test('Can insert text content', async t => {
+  const { page } = t.context
+
+  await page.click(test_selector)
+
+  await page.keyboard.type('foo')
+
+  t.true((await page.$eval(test_selector, el => el.innerHTML)).includes('foo'))
```
Comment (argyleink) [inline app/features/text.test.js]: 😎

File: app/features/position.test.js (hunk 1 above)
Comment (dgeare) [reply app/features/position.test.js]: (•_•)
( •_•)>⌐■-■
(⌐■_■)

**Pull request merged**